x-pack/filebeat/input/httpjson/request.go

```diff
@@ -276,37 +283,65 @@ func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, p
 		urlCopy           url.URL
 		urlString         string
 		httpResp          *http.Response
+		initialResponse   []*http.Response
 		intermediateResps []*http.Response
 		finalResps        []*http.Response
+		isChainExpected   bool
+		chainIndex        int
 	)
+
 	for i, rf := range r.requestFactories {
 		finalResps = nil
 		intermediateResps = nil
 		// iterate over collected ids from last response
 		if i == 0 {
 			// perform and store regular call responses
-			httpResp, err = rf.collectResponse(stdCtx, trCtx, r) //nolint:bodyclose // Bad linter! The response body will always be closed by drainBody function.
+			httpResp, err = rf.collectResponse(stdCtx, trCtx, r)
 			if err != nil {
 				return fmt.Errorf("failed to execute rf.collectResponse: %w", err)
 			}
 			if len(r.requestFactories) == 1 {
-				finalResps = append(finalResps, httpResp) //nolint:bodyclose // Bad linter! The response body will always be closed by drainBody function.
-				n = r.processAndPublishEvents(stdCtx, trCtx, publisher, finalResps, true, i)
+				finalResps = append(finalResps, httpResp)
+				n = processAndPublishEvents(trCtx, publisher, true, r.log, r.responseProcessors[i].startProcessing(stdCtx, trCtx, finalResps))
```

Comment:
With the change noted above.
```suggestion
				events := r.responseProcessors[i].startProcessing(stdCtx, trCtx, finalResps)
				n = processAndPublishEvents(trCtx, events, publisher, true, r.log)
```